### PR TITLE
fix replay pos in TransformReplay::replayCasP

### LINKED ISSUE #2593: bug in propagateReshapeTransforms
To reproduce use: [repro](https://github.com/NVIDIA/Fuser/tree/llu/bug_view)
The following fusion failed with err msg `C++ exception with description "maybe_unmapped_ids.count(p_id) INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/transform_replay.cpp":603, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Could not find axis, iS34{( ceilDiv(i0, 3) )}rf, requested in replaying consumer T6_g[ iS44{3}, iS45{( ( ceilDiv(i0, 3) ) * 4 )}, iS46{( ceilDiv(i2, 4) )} ] as producer T9_l[ iS30{3}, iS33{( ceilDiv(i2, 4) )}, iS34{( ceilDiv(i0, 3) )}rf, iS35{4}rf ]
Exception raised from replayCasP at /opt/pytorch/nvfuser/csrc/transform_replay.cpp:603`
```

TEST_F(GpuViewTest, SplitMergePointwiseSplitMerge) {
  auto fusion = std::make_unique<Fusion>();
  FusionGuard fg(fusion.get());
  const std::vector<int64_t> input_shape = {12, 20};
  DataType dtype = DataType::Float;
  auto tv0 = makeContigTensor(input_shape.size(), dtype);
  fusion->addInput(tv0);
  auto tv1 = castOp(DataType::Float, tv0);
  // root domain : (i0, i2)
  // logi domain : (3, i0/3, 4, i2/4)
  auto tv2 = reshape(tv1, {12, 20}, {3, 4, 4, 5});
  // root domain : (3, i0/3, 4, i2/4)
  // logi domain : (3, i0/3*4, i2/4)
  auto tv3 = reshape(tv2, {3, 4, 4, 5}, {3, 16, 5});
  // root domain : (3, i0/3*4, i2/4)
  auto tv4 = mul(tv3, tv3);
  // root domain : (i0, i2)
  // logi domain : (3, i0/3, 4, i2/4)
  auto tv5 = reshape(tv1, {12, 20}, {3, 4, 4, 5});
  // root domain : (3, i0/3, 4, i2/4)
  // logi domain : (3, i0/3*4, i2/4)
  auto tv6 = reshape(tv5, {3, 4, 4, 5}, {3, 16, 5});
  fusion->addOutput(tv4);
  fusion->addOutput(tv6);

  auto options =
      at::TensorOptions().dtype(data_type_to_aten(dtype)).device(at::kCUDA, 0);
  auto t0 = at::randn(input_shape, options);

  FusionExecutorCache executor_cache(std::move(fusion));
  auto cg_outputs = executor_cache.runFusionWithInputs({t0});
}
```

Some debug info
```
Propagating transformations from T2_l[ iS7{3}rf, iS11{( ceilDiv(i2, 4) )}rf, iS8{( ceilDiv(i0, 3) )}rf, iS10{4}rf ]
Error in replayCasP consumer T6_g[ iS44{3}, iS45{( ( ceilDiv(i0, 3) ) * 4 )}, iS46{( ceilDiv(i2, 4) )} ]
 logical domain : (iS44{3}, iS45{( ( ceilDiv(i0, 3) ) * 4 )}, iS46{( ceilDiv(i2, 4) )})
 allocation domain : (iS44{3}, iS45{( ( ceilDiv(i0, 3) ) * 4 )}, iS46{( ceilDiv(i2, 4) )})
 contiguity: t t t
 loop domain : (iS44{3}, iS45{( ( ceilDiv(i0, 3) ) * 4 )}, iS46{( ceilDiv(i2, 4) )})

Error in replayCasP consumer T9_l[ iS30{3}, iS33{( ceilDiv(i2, 4) )}, iS34{( ceilDiv(i0, 3) )}rf, iS35{4}rf ]
 root domain : (iS30{3}, iS34{( ceilDiv(i0, 3) )}rf, iS35{4}rf, iS33{( ceilDiv(i2, 4) )})
  Merge: iS34{( ceilDiv(i0, 3) )}rf and iS35{4}rf -> iS36{( ( ceilDiv(i0, 3) ) * 4 )}rf
 logical domain : (iS30{3}, iS36{( ( ceilDiv(i0, 3) ) * 4 )}rf, iS33{( ceilDiv(i2, 4) )})
 allocation domain : (iS30{3}, iS36{( ( ceilDiv(i0, 3) ) * 4 )}rf, iS33{( ceilDiv(i2, 4) )})
 contiguity: t t t
 loop domain : (iS30{3}, iS33{( ceilDiv(i2, 4) )}, iS34{( ceilDiv(i0, 3) )}rf, iS35{4}rf)
```

### PR DESCRIPTION
**Pull request by liqiangxl**
Fix #2593 
```
  // pay attention to consumer pos, it should be the actual replayed axis,
  // otherwise further replay may lead to error where the mapped ID is missing.
  // see https://github.com/NVIDIA/Fuser/issues/2593
```

### CODE REVIEW

**Comment by liqiangxl** [comment]
!build --diff

**Comment by liqiangxl** [comment]
Failed cases in `nvfuser-ci/jit_python_tests_17_A100` is due to out of memory.
`/jit_thunder_tests_17` also failed in other CIs, not related to this PR.

**Comment by liqiangxl** [comment]
!build

File: csrc/transform_replay.cpp [hunk 1]
```diff
@@ -701,6 +701,11 @@ std::pair<TensorDomain*, int64_t> TransformReplay::replayCasP(
     used_IDs.emplace(it->second);
   }
 
+  // pay attention to consumer pos, it should be the actual replayed axis,
+  // otherwise further replay may lead to error where the mapped ID is missing.
+  // see https://github.com/NVIDIA/Fuser/issues/2593
```
Comment (jjsjann123) [inline csrc/transform_replay.cpp:706]: nitpick.

Totally not your problem. But can we have a follow up PR to fix the comment above at [here]( https://github.com/NVIDIA/Fuser/blob/8f974c8d39af9851bacb34af08c10287f3e9c077/csrc/transform_replay.cpp#L664-L685). I think it's copied from [here](https://github.com/NVIDIA/Fuser/blob/8f974c8d39af9851bacb34af08c10287f3e9c077/csrc/transform_replay.cpp#L433-L453), but we forgot to rename things.

Even better, would be to merge the comment in a single place and reference it in the other place.

File: tests/cpp/test_gpu_view.cpp [hunk 2]
```diff
@@ -2344,4 +2344,37 @@ TEST_F(GpuViewTest, ExpandedBroadcast) {
...
+TEST_F(GpuViewTest, SplitMergePointwiseSplitMerge) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  const std::vector<int64_t> input_shape = {12, 20};
+  DataType dtype = DataType::Float;
+  auto tv0 = makeContigTensor(input_shape.size(), dtype);
+  fusion->addInput(tv0);
+  auto tv1 = castOp(DataType::Float, tv0);
+  // root domain : (i0, i2)
+  // logi domain : (3, i0/3, 4, i2/4)
```
Comment (jjsjann123) [inline tests/cpp/test_gpu_view.cpp:2356]: nitpick: logical domain?

File: tests/cpp/test_gpu_view.cpp [hunk 3]
```diff
@@ -2344,4 +2344,37 @@ TEST_F(GpuViewTest, ExpandedBroadcast) {
...
+  auto tv6 = reshape(tv5, {3, 4, 4, 5}, {3, 16, 5});
+  fusion->addOutput(tv4);
+  fusion->addOutput(tv6);
+
+  auto options =
+      at::TensorOptions().dtype(data_type_to_aten(dtype)).device(at::kCUDA, 0);
+  auto t0 = at::randn(input_shape, options);
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+  auto cg_outputs = executor_cache.runFusionWithInputs({t0});
```
Comment (jjsjann123) [inline tests/cpp/test_gpu_view.cpp:2377]: Are we missing the validation?

**Review by jjsjann123 — COMMENTED** [review]
let's fix the test with validation. Everything looks good to me.

File: tests/cpp/test_gpu_view.cpp (hunk 2 above)
Comment (liqiangxl) [reply tests/cpp/test_gpu_view.cpp]: yes, I used `logi` since it has the same length with `root`, will change to `logical` to avoid confusion.

**Review by jjsjann123 — APPROVED** [review]
stamping since this looks straightforward to me.

**Review by naoyam — APPROVED** [review]
LGTM as well.

**Pull request merged**